### ISSUE TITLE
regex exception thrown in MBody1 GPU test case

### ISSUE THREAD

**Issue by sharkovsky** [genn-team/genn#145]
Hello, 

I get an exception error when executing the tests. Specifically I get a `terminate called after throwing an instance of 'std::regex_error'` while executing the `MBody1 GPU` test. 

I am using commit 1ffc805d239d (latest master as of now), gcc version 4.9.0 and cuda 7.5, on an x86 machine + Tesla K20m gpu.

The full output from executing `userproject/testprojects.sh` is shown below.

```
This script may take a while to finish all tests if you use it with default parameters. 
You may consider shortening the duration of simulation in classol_sim.h for the MBody examples.

Making tools...
rm -rf *.o *.dSYM gen_input_structured gen_pnlhi_syns gen_kcdn_syns gen_kcdn_syns_fixto10K gen_pnkc_syns gen_pnkc_syns_indivID gen_syns_sparse gen_syns_sparse_izhModel 
g++ -Wall -Winline -O3 -std=c++11 -o gen_input_structured gen_input_structured.cc -I"/gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/userproject/include"
g++ -Wall -Winline -O3 -std=c++11 -o gen_pnlhi_syns gen_pnlhi_syns.cc -I"/gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/userproject/include"
g++ -Wall -Winline -O3 -std=c++11 -o gen_kcdn_syns gen_kcdn_syns.cc -I"/gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/userproject/include"
g++ -Wall -Winline -O3 -std=c++11 -o gen_kcdn_syns_fixto10K gen_kcdn_syns_fixto10K.cc -I"/gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/userproject/include"
g++ -Wall -Winline -O3 -std=c++11 -o gen_pnkc_syns gen_pnkc_syns.cc -I"/gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/userproject/include"
g++ -Wall -Winline -O3 -std=c++11 -o gen_pnkc_syns_indivID gen_pnkc_syns_indivID.cc -I"/gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/userproject/include"
g++ -Wall -Winline -O3 -std=c++11 -o gen_syns_sparse gen_syns_sparse.cc -I"/gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/userproject/include"
g++ -Wall -Winline -O3 -std=c++11 -o gen_syns_sparse_izhModel gen_syns_sparse_izhModel.cc -I"/gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/userproject/include"
firstrun  false

TEST 1: TEST BY CREATING NEW INPUT PATTERNS

*******************************************************************************


*********************** Testing MBody1 ****************************************


*******************************************************************************
Checking if benchmarking directory exists...
benchmarking directory exists. Using input data from /gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/userproject/refProjFiles/MBody1

rm -f generate_run
cd ../tools && make clean
make[1]: Entering directory `/gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/userproject/tools'
rm -rf *.o *.dSYM gen_input_structured gen_pnlhi_syns gen_kcdn_syns gen_kcdn_syns_fixto10K gen_pnkc_syns gen_pnkc_syns_indivID gen_syns_sparse gen_syns_sparse_izhModel 
make[1]: Leaving directory `/gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/userproject/tools'
g++ -Wall -Winline -O3 -o generate_run generate_run.cc -I"/gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/lib/include" -I"/gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/userproject/include"
cd ../tools && make
make[1]: Entering directory `/gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/userproject/tools'
g++ -Wall -Winline -O3 -std=c++11 -o gen_input_structured gen_input_structured.cc -I"/gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/userproject/include"
g++ -Wall -Winline -O3 -std=c++11 -o gen_pnlhi_syns gen_pnlhi_syns.cc -I"/gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/userproject/include"
g++ -Wall -Winline -O3 -std=c++11 -o gen_kcdn_syns gen_kcdn_syns.cc -I"/gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/userproject/include"
g++ -Wall -Winline -O3 -std=c++11 -o gen_kcdn_syns_fixto10K gen_kcdn_syns_fixto10K.cc -I"/gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/userproject/include"
g++ -Wall -Winline -O3 -std=c++11 -o gen_pnkc_syns gen_pnkc_syns.cc -I"/gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/userproject/include"
g++ -Wall -Winline -O3 -std=c++11 -o gen_pnkc_syns_indivID gen_pnkc_syns_indivID.cc -I"/gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/userproject/include"
g++ -Wall -Winline -O3 -std=c++11 -o gen_syns_sparse gen_syns_sparse.cc -I"/gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/userproject/include"
g++ -Wall -Winline -O3 -std=c++11 -o gen_syns_sparse_izhModel gen_syns_sparse_izhModel.cc -I"/gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/userproject/include"
make[1]: Leaving directory `/gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/userproject/tools'


####################### MBody1 GPU ######################
cd model && genn-buildmodel.sh MBody1.cc && make clean all SIM_CODE=MBody1_CODE
ar -rcs /gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/lib/lib/libgenn.a /gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/lib/obj/global.o /gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/lib/obj/modelSpec.o /gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/lib/obj/neuronModels.o /gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/lib/obj/synapseModels.o /gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/lib/obj/postSynapseModels.o /gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/lib/obj/utils.o /gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/lib/obj/stringUtils.o /gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/lib/obj/sparseUtils.o /gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/lib/obj/hr_time.o
g++ -std=c++11 -DNVCC=\"""/gpfs/bbp.cscs.ch/apps/viz/bbp/dev/cuda/7.5/"/bin/nvcc"\" -DMODEL=\"/gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/userproject/MBody1_project/model/MBody1.cc\" -o /gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/userproject/MBody1_project/model/generateALL /gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/lib/src/generate*.cc -I"/gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/lib/include" -I""/gpfs/bbp.cscs.ch/apps/viz/bbp/dev/cuda/7.5/"/include" -L"/gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/lib/lib" -L""/gpfs/bbp.cscs.ch/apps/viz/bbp/dev/cuda/7.5/"/lib64" -L""/gpfs/bbp.cscs.ch/apps/viz/bbp/dev/cuda/7.5/"/lib64/stubs" -lgenn -lcuda -lcudart
call was ./generateALL /gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/userproject/MBody1_project/model 
optimizing block size...
Global memory required for core model: 1.24618 MB. 
4972412928 for device 0
terminate called after throwing an instance of 'std::regex_error'
  what():  regex_error
/gpfs/bbp.cscs.ch/project/proj16/cremones/NIPS-Landscape-paper/bench/GeNN/genn/lib/bin/genn-buildmodel.sh: line 76: 10145 Aborted                 (core dumped) "$GENERATEALL" "$OUT_PATH"
genn-buildmodel.sh:75: error 50: command failure
ERROR: Following call failed with status 12800:
cd model && genn-buildmodel.sh MBody1.cc && make clean all SIM_CODE=MBody1_CODE
Exiting...
```

Thank you

**Comment by tnowotny** [comment]
Thanks for your report. We haven't seen this before - we will look into it ...

**Comment by neworderofjamie** [comment]
Hi there,

I have investigated this issue and it seems that, while we thought regular expressions should work properly in GCC 4.9.0, a GCC bug means that GCC 4.9.1 is actually required. I have made a fix in #146 but, unfortunately the oldest version of GCC I can convince my computer to install is 4.9.3 so I cannot actually test this. Could you try the branched version and let us know if it solves your issue?

Thanks

Jamie

**Comment by sharkovsky** [comment]
Thank you for your fast reply!

I can confirm that the fix in #146 worked for me, I am now able to run MBody1 GPU test. 

Thank you for your support.
